dashboard/test/controllers/api/v1/sections_controller_test.rb

```diff
@@ -95,4 +95,39 @@ class Api::V1::SectionsControllerTest < ActionController::TestCase
 
     assert_equal @course.id, json['course_id']
   end
+
+  test "join with invalid section code" do
+    assert_raises(ActiveRecord::RecordNotFound) do
+      post :join, params: {id: 'xxxxxx'}
+    end
+  end
+
+  test "join with nobody signed in" do
+    post :join, params: {id: @word_section.code}
+    assert_response 404
+  end
+
+  test "join with valid section code" do
+    student = create :student
+    sign_in student
+    post :join, params: {id: @word_section.code}
+    assert_response :success
+  end
+
+  test "leave with invalid section code" do
+    assert_raises(ActiveRecord::RecordNotFound) do
+      post :leave, params: {id: 'xxxxxx'}
+    end
+  end
+
+  test "leave with nobody signed in" do
+    post :leave, params: {id: @word_section.code}
+    assert_response 404
+  end
+
+  test "leave with valid section code" do
+    sign_in @word_user_1
+    post :leave, params: {id: @word_section.code}
+    assert_response :success
+  end
```

Comment:
Other test cases you could add:
- Join a section I'm already a member of
- Leave a real section that I'm not a member of

Is there a case we should check where a user doesn't have permission to leave a section, or are we only enforcing that on the client?